### PR TITLE
feat(avro-to-json): add wireFormat configuration parameter

### LINKED ISSUE #337: feat(avro-to-json): add wireFormat configuration parameter for deterministic Avro parsing
### Summary

Application: `src/500-application/512-avro-to-json`
WASM Operator for Azure IoT Operations

Add an optional `wireFormat` configuration parameter to the `avro-to-json` operator that lets users declare the wire format of incoming Avro messages. This replaces the current implicit try/fallback approach with deterministic, format-specific parsing when the format is known, while preserving backward-compatible auto-detection as the default.

### Motivation

The current `parse_with_schema` function attempts raw Avro parsing first, and on failure retries after stripping a 5-byte Confluent Schema Registry prefix. This has two problems:

1. **Ambiguity** — `strip_confluent_header` triggers on any payload starting with `0x00` longer than 5 bytes. If raw Avro data legitimately starts with `0x00` and fails for an unrelated reason (schema mismatch, truncation), the retry produces a misleading error.
2. **Extensibility** — other schema registries (Apicurio, Karapace) use different header layouts. Adding more fallback branches creates a fragile chain of guesses.

A `wireFormat` parameter makes parsing deterministic when the format is known, and keeps the `auto` fallback for users who prefer zero-config.

### Configuration Values

| Value       | Behavior                                                                           |
|-------------|------------------------------------------------------------------------------------|
| `auto`      | Default. Current try-raw-then-strip-and-retry logic (backward-compatible).         |
| `confluent` | Always strip 5-byte Confluent prefix (`0x00` + 4-byte schema ID) before parsing.  |
| `raw`       | Parse directly with no prefix handling. Fail immediately if parsing fails.         |

#### Rust Code (`operators/avro-to-json/src/lib.rs`)

- Add `WireFormat` enum (`Auto`, `Confluent`, `Raw`) and a `WIRE_FORMAT` `OnceLock` static
- Parse `wireFormat` from configuration properties in `avro_init`
- Refactor `parse_with_schema` to dispatch on `WireFormat` instead of implicit try/fallback
- Extract `parse_with_schema_inner(data, schema, wire_format)` for testability
- Keep `strip_confluent_header` unchanged
- Add ~7 unit tests covering each variant

#### Graph YAML

- Add `wireFormat` parameter to `moduleConfigurations` in `resources/graphs/graph-avro-to-json.yaml`

#### Documentation

- Update configuration table in `src/500-application/512-avro-to-json/README.md`
- Add "Wire Format" section explaining the three options
- Update Avro Format Detection priority table
- Add troubleshooting entry for Confluent prefix errors
- Update `blueprints/full-single-node-cluster/terraform/dataflow-graphs-avro-json.tfvars.example`

### Files to Modify

| File | Change |
|------|--------|
| `operators/avro-to-json/src/lib.rs` | Add `WireFormat` enum, static, init parsing, refactor `parse_with_schema`, add tests |
| `resources/graphs/graph-avro-to-json.yaml` | Add `wireFormat` parameter definition |
| `blueprints/full-single-node-cluster/terraform/dataflow-graphs-avro-json.tfvars.example` | Add `wireFormat` configuration entry |
| `src/500-application/512-avro-to-json/README.md` | Config table, Wire Format section, detection table, troubleshooting |

### Version Impact

Non-breaking, additive change. Default `auto` behavior matches the current implementation exactly. Existing deployments require no configuration changes. Bump minor version in `Cargo.toml`.

### PR DESCRIPTION
**Pull request by katriendg**
## Description

Added an optional **wireFormat** configuration parameter to the *avro-to-json* WASM operator, enabling deterministic Avro parsing when the wire format of incoming messages is known. The parameter accepts three values — `auto`, `confluent`, and `raw` — and defaults to `auto`, which preserves the existing try-raw-then-retry behavior for full backward compatibility.

### Rust Wasm Operator

The core change introduced a `WireFormat` enum and a `WIRE_FORMAT` `OnceLock` static in *lib.rs*. The existing `parse_with_schema` function was refactored: a new **`parse_with_schema_inner`** function accepts the wire format explicitly and dispatches to the correct parsing path. The original `parse_with_schema` now delegates to the inner function using the configured format, keeping the public API unchanged.

Configuration parsing in `avro_init` reads the `wireFormat` property with case-insensitive matching, logs a warning for unknown values (defaulting to `Auto`), and logs the selected format at `Info` level for observability.

Seven unit tests cover all three variants — raw success/failure, confluent success/failure, auto success on both raw and confluent data, and a short-payload edge case for confluent mode.

### Graph YAML and Blueprint

The *graph-avro-to-json.yaml* gained a `wireFormat` parameter declaration in `moduleConfigurations`. The blueprint *dataflow-graphs-avro-json.tfvars.example* added a `wireFormat` key-value entry with inline documentation of all three options, demonstrating the Confluent use case.

### Version

Package version bumped from 1.0.0 to **1.1.0** in both *Cargo.toml* and *Cargo.lock*.

## Related Issue

Fixes #337

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Blueprint modification or addition
- [x] Component modification or addition
- [x] Documentation update
- [ ] CI/CD pipeline change
- [ ] Other (please describe):

## Implementation Details

- Introduced `WireFormat` enum (`Auto`, `Confluent`, `Raw`) with `OnceLock`-based static initialization
- Extracted `parse_with_schema_inner(data, schema, wire_format)` from `parse_with_schema` for testable, format-specific dispatch
- `avro_init` parses the `wireFormat` configuration property with case-insensitive matching and defaults to `Auto` for unknown or missing values
- Error messages are mode-specific for production debugging (e.g., Confluent mode reports missing 5-byte prefix with payload size)
- Graph YAML declares the new parameter for operator discovery
- Blueprint tfvars example documents all three options with the Confluent use case

## Testing Performed

- [ ] Terraform plan/apply
- [ ] Blueprint deployment test
- [x] Unit tests
- [ ] Integration tests
- [ ] Bug fix includes regression test (see [Test Policy](docs/contributing/testing-validation.md))
- [x] Manual validation
- [ ] Other:

## Validation Steps

1. Review the `WireFormat` enum and `parse_with_schema_inner` dispatch logic in *lib.rs*
2. Run `cargo test` in `src/500-application/512-avro-to-json/operators/avro-to-json/` — all 7 new wireFormat tests should pass
3. Verify the graph YAML declares `wireFormat` in `moduleConfigurations`
4. Verify the blueprint tfvars example includes the `wireFormat` key-value entry
5. Confirm version bump to 1.1.0 in *Cargo.toml*
6. Run a blueprint deployment with the updated graph and validate correct parsing behavior for each wire format option (requires appropriate test messages for raw and Confluent formats)

## Checklist

- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] I have run `terraform fmt` on all Terraform code
- [ ] I have run `terraform validate` on all Terraform code
- [ ] I have run `az bicep format` on all Bicep code
- [ ] I have run `az bicep build` to validate all Bicep code
- [x] I have checked for any sensitive data/tokens that should not be committed
- [x] Lint checks pass (run applicable linters for changed file types)

## Security Review

- [x] No credentials, secrets, or tokens are hardcoded or logged
- [x] RBAC and identity changes follow least-privilege principles
- [x] No new network exposure or public endpoints introduced without justification
- [x] Dependency additions or updates have been reviewed for known vulnerabilities
- [x] Container image changes use pinned digests or SHA references

## Additional Notes

This is a non-breaking, additive change. The default `auto` behavior matches the current implementation exactly. Existing deployments require no configuration changes.

### CODE REVIEW

**Review by nguyena2 — APPROVED** [review]

File: src/500-application/512-avro-to-json/operators/avro-to-json/src/lib.rs [hunk 1]
```diff
@@ -817,4 +891,121 @@ mod tests {
...
+    fn parse_inner_confluent_mode_short_payload_fails() {
+        let schema_json = r#"{"type":"record","name":"Msg","fields":[
+            {"name":"a","type":"string"}
+        ]}"#;
+        let schema = Schema::parse_str(schema_json).unwrap();
+        let short = vec![0x00, 0x01, 0x02];
+        let result = parse_with_schema_inner(&short, &schema, WireFormat::Confluent);
+        assert!(result.is_err());
+        assert!(result.unwrap_err().contains("does not have the expected"));
+    }
```
Comment (auyidi1) [inline src/500-application/512-avro-to-json/operators/avro-to-json/src/lib.rs:1010]: nit: No tests for `avro_init` configuration parsing.

All 7 wireFormat tests call `parse_with_schema_inner` directly. There's no unit test coverage for:

- Case-insensitive matching — e.g., `"CONFLUENT"`, `"Confluent"`, `"  raw  "` (trim + lowercase)
- Unknown value defaulting — e.g., `"invalid" → Auto` with warning
- Missing wireFormat — defaults to `Auto`
- `OnceLock::set` behavior — the `let _ = WIRE_FORMAT.set(...)` silently ignores errors if called twice

I acknowledge integration testing may catch some of these, but a unit test for the config parsing logic would be valuable given the `trim().to_lowercase()` transformations.

Comment (WilliamBerryiii) [reply src/500-application/512-avro-to-json/operators/avro-to-json/src/lib.rs:1010]: I'll file an issue for this so we can get this merged.

Comment (WilliamBerryiii) [reply src/500-application/512-avro-to-json/operators/avro-to-json/src/lib.rs:1010]: Filed https://github.com/microsoft/edge-ai/issues/359 to track the missing avro_init configuration parsing tests.Filed https://github.com/microsoft/edge-ai/issues/359 to track the missing avro_init configuration parsing tests.
